library/packages/src/lib/y2packager/licenses_fetchers/rpm.rb

```diff
@@ -0,0 +1,154 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2018 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "y2packager/licenses_fetchers/base"
+
+module Y2Packager
+  module LicensesFetchers
+    # This class is responsible for obtaining the license and license content
+    # of a given product from a RPM package
+    class Rpm < Base
+      # Return the license text to be confirmed
+      #
+      # @param lang [String] Language
+      #
+      # @return [String, nil] Product's license; nil if the product or the license were not found
+      def content(lang)
+        super
+
+        if package.nil?
+          log.info("No package found for #{product_name}")
+
+          return nil
+        end
+
+        license_content_for(lang)
+      end
+
+      # Return available locales for product's license
+      #
+      # @return [Array<String>] Language codes ("de_DE", "en_US", etc.)
+      def locales
+        if package.nil?
+          log.error "Error getting license translations: no package found for #{product_name}"
+
+          return []
+        end
+
+        @locales ||=
+          begin
+            tmpdir = Dir.mktmpdir
+            package.extract_to(tmpdir)
+            # TODO: Use rpm -qpl file.rpm instead?
+            license_files = Dir.glob(File.join(tmpdir, "**", "LICENSE.*.TXT"), File::FNM_CASEFOLD)
+            languages = license_files.map { |path| path[/LICENSE.(\w*).TXT/i, 1] }
+            languages << DEFAULT_LANG
+            languages.compact.uniq
+          end
+      end
+
+    private
+
+      # Return the license content for a package and language
+      #
+      # Package is extracted to a work directory. When a license for a language "xx_XX" is not
+      # found, it will fallback to "xx".
+      #
+      # @see license_file
+      #
+      # @param package [Y2Packager::Package] Product package
+      # @param lang    [String] Searched language
+      #
+      # @return [Array<String, String>, nil] Array containing content and language code
+      def license_content_for(lang)
+        tmpdir = Dir.mktmpdir
+
+        begin
+          package.extract_to(tmpdir)
+          license_file = license_path(tmpdir, lang) || fallback_path(tmpdir)
+
+          if license_file.nil?
+            log.error("License file not found")
```

Comment:
if you write error to log try to write for which license file was license was not found.

Reply:
At this point, the neither license file was found, file for requested lang/translation nor fallback file. Do you want to add the `lang` to the error message?

If you take a look to [`license_path`](https://github.com/yast/yast-yast2/blob/ca4665afee0c806fb619b0541c6c51c40b6bd4a8/library/packages/src/lib/y2packager/licenses_fetchers/rpm.rb#L99) and [`fallback_path`](https://github.com/yast/yast-yast2/blob/ca4665afee0c806fb619b0541c6c51c40b6bd4a8/library/packages/src/lib/y2packager/licenses_fetchers/rpm.rb#L115) methods, you can see that we are already loggin infirmation about what we are looking for. Isn't it enough?

Reply:
well, I think it is better to have all important info in that error call. So when you searching logs for errors, you see it immediatelly. It is important especially when multiple licenses are needed like in SLE with several addons